src/job_cache/eviction_policy.cpp

```diff
@@ -68,6 +68,12 @@ struct TTLEvictionPolicyImpl {
 
     wcl::log::info("Performing a cleanup!")();
 
+    // We're about to assign to this thread but we don't want to
+    // wait for it to exit so we detach the thread before making a new one.
+    // The function being run is thread safe so many instances can be run at
+    // once safely.
+    if (cleaning_thread.joinable()) cleaning_thread.detach();
```

Comment:
Do we need to keep track of the detached threads in anyway? My reading of the documentation is that all resources will be freed once the detached thread exits, so we only need to be sure that `remove_backing_files` eventually completes, which seems reasonable.

Reply:
Correct